### PR TITLE
Support slash as a command trigger

### PR DESCRIPTION
**Pull request by neolit123**
me and Martin collaborated on this change over a couple of Zoom meetings.
please, let me know if this needs any changes.
/cc @rgee0 @alexellis 

## Description

Track the environment variable `use_slash_trigger`
and make it possible to use `/` as the command trigger
that the bot uses instead of "Derek ".

- Update unit tests in commentHandler_test.go.
- Update README.md

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>
Signed-off-by: Lubomir I. Ivanov (VMware) <neolit123@gmail.com>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md))
https://github.com/alexellis/derek/issues/63

## How Has This Been Tested?

unit tests only on my side.
@mrtind and @ivanayov can possibly do an e2e if they have the setup.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rgee0** [comment]
Not sure about `DEREKCOMMANDPREFIX`.  How does it compare to our existing env-vars?

**Comment by neolit123** [comment]
i'm open to suggestions of the exact naming. :)

**Comment by rgee0** [comment]
If it stays as it is then it should follow the existing pattern `derek_command_prefix`.  I wonder whether `trigger` or, `trigger_value`, or `trigger_term` or `invocation_value`, etc might be intuitive for the user.

File: commentHandler.go [hunk 1]
```diff
@@ -30,6 +30,16 @@ const (
 	addLabelConstant    string = "AddLabel"
 )
 
+const defaultCommandPrefix = "Derek "
+var commandPrefix = ""
+
+func init() {
```
Comment (alexellis) [inline commentHandler.go]: Let's move this into the function handler entry-point instead.

**Comment by neolit123** [comment]
> If it stays as it is then it should follow the existing pattern derek_command_prefix. I wonder whether trigger or, trigger_value, or trigger_term or invocation_value, etc might be intuitive for the user.

@alexellis any comments on the naming on the env variable?

**Comment by alexellis** [comment]
My idea is: `bot_name` with a default of Derek when not specified.

File: commentHandler.go (hunk 1 above)
Comment (alexellis) [reply commentHandler.go]: @neolit123

File: README.md [hunk 2]
```diff
@@ -53,6 +53,11 @@ redirect: https://github.com/<some-user>/<some-repo>/.DEREK.yaml
 If this optional field is non-empty, Derek will read it's configuration from another location. This allows multiple projects to use the same configuration.
 Please note that redirection is only supported for GitHub repository URLs.
 
+* Custom command prefixes
+
+By default, Derek commands can be called with `Derek <some-command>`. The prefix `Derek ` is the default value and this can be changed by using the `DEREKCOMMANDPREFIX` environment variable.
```
Comment (alexellis) [inline README.md]: @neolit123 I vote for `derek_bot_name` or `bot_name` .. @rgee0 wdyt?

Comment (rgee0) [reply README.md]: I think we should go with something from which the user can infer meaning.  Interesting that the issue covering this refers to a `trigger` and this pull request talks of a `command_prefix`.  These are terms that have naturally emerged as we've thought about the nature of the feature.

Derek should still be Derek, whether you shout his name, ring a bell, or make lights flash in order to attract his attention

Comment (neolit123) [reply README.md]: regarding `bot_name`, i should remind that the default value is not only `Derek`, but it's `Derek ` (with an extra space after the name).

**Comment by neolit123** [comment]
> Derek should still be Derek, whether you shout his name, ring a bell, or make lights flash in order to attract his attention

i've already said this in another comment, but if identity preservation is a project goal, i suggest that we close this PR and the related issue.

**Comment by rgee0** [comment]
The change is required.  

The point was/is that how the bot is invoked should be separate to what the bot is called.  If someone want to use `/` or `~` or  `%` to trigger Derek then they should be able to.  In that case I would see `/` or `~` or  `%` as the trigger for Derek, not Derek's new name.

**Comment by neolit123** [comment]
ok, waiting on decisions for:
- env. variable naming.
- `trigger` vs `command prefix` usage in code and docs.

**Comment by alexellis** [comment]
Bump, we also have merge conflicts now.

**Comment by neolit123** [comment]
> ok, waiting on decisions for

**Comment by alexellis** [comment]
I suggested `derek_bot_name` as an environmental variable. Can that work? We also need a rebase on the PR please.

**Comment by neolit123** [comment]
> I suggested derek_bot_name as an environmental variable. Can that work? 

whatever you and @rgee0 would prefer.

1) env. variable naming.
2) `trigger` vs `command prefix` usage in code and docs.

please provide decisions on the above two points and we can proceed with a rebase.

**Comment by neolit123** [comment]
@alexellis 

> Let's have Derek always called Derek and the slash command being only "Derek " or "/" cc @neolit123 does this help us move forward with a rebase/close? @rgee0

updated, PTAL.

**Comment by neolit123** [comment]
@rgee0 let me try that quickly.

**Comment by martindekov** [comment]
Ok @neolit123 thanks I will be taking a look for further comments

**Comment by neolit123** [comment]
@rgee0 updated, PTAL.

**Comment by alexellis** [comment]
Merged. Thank you all

**Pull request merged**